drivers/timer/ace_v1x_timer.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/device.h>
+#include <zephyr/drivers/timer/system_timer.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/spinlock.h>
+#include <zephyr/drivers/interrupt_controller/dw_ace_v1x.h>
+
+#include <cavs-idc.h>
+#include <cavs-shim.h>
+#include <ace_v1x-regs.h>
+
+/**
+ * @file
+ * @brief ACE 1.x DSP Wall Clock Timer driver
+ *
+ * The ACE 1.x DSP on Intel SoC has a timer with one counter and two compare
+ * registers that is external to the CPUs. This timer is accessible from
+ * all available CPU cores and provides a synchronized timer under SMP.
+ */
+
+#define COMPARATOR_IDX  0 /* 0 or 1 */
+
+#ifdef CONFIG_SOC_SERIES_INTEL_ACE1X
+#define TIMER_IRQ MTL_IRQ_TO_ZEPHYR(MTL_INTL_TTS)
+#else
+#define TIMER_IRQ DSP_WCT_IRQ(COMPARATOR_IDX)
+#endif
+
+#define CYC_PER_TICK	(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC	\
+			/ CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+#define MAX_CYC		0xFFFFFFFFUL
+#define MAX_TICKS	((MAX_CYC - CYC_PER_TICK) / CYC_PER_TICK)
+#define MIN_DELAY	(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC / 100000)
+
+#define WCTCS_TTIE(c) BIT(8 + (c))
+
+BUILD_ASSERT(MIN_DELAY < CYC_PER_TICK);
+BUILD_ASSERT(COMPARATOR_IDX >= 0 && COMPARATOR_IDX <= 1);
+
+#ifdef CONFIG_SOC_SERIES_INTEL_ACE1X
+/* Basically identical register interface, very slightly different layout */
+# define WCTCS      (&MTL_TTS.wctcs)
+# define COUNTER_HI (&MTL_TTS.wc.hi)
+# define COUNTER_LO (&MTL_TTS.wc.lo)
+# define COMPARE_HI (&MTL_TTS.wctc[COMPARATOR_IDX].hi)
+# define COMPARE_LO (&MTL_TTS.wctc[COMPARATOR_IDX].lo)
+#else
+# define WCTCS      (&CAVS_SHIM.dspwctcs)
+# define COUNTER_HI (&CAVS_SHIM.dspwc_hi)
+# define COUNTER_LO (&CAVS_SHIM.dspwc_lo)
+# define COMPARE_HI (&CAVS_SHIM.UTIL_CAT(UTIL_CAT(dspwct, COMPARATOR_IDX), c_hi))
+# define COMPARE_LO (&CAVS_SHIM.UTIL_CAT(UTIL_CAT(dspwct, COMPARATOR_IDX), c_lo))
+#endif
```

Comment:
Still cutting and pasting here.  I note (heh) that the blocks above implement drivers for both ACE and cAVS hardware via some slightly kludgey pointer macros, and that the rest of the file looks essentially identical to the pre-existing cavs_timer.c in the same directory.  I'm guessing that if you drop this in place on top of cavs_timer.c and have the MTL board enable CONFIG_CAVS_TIMER (and no doubt fix up a few collisions and bitrot spots), the same driver will work for the whole hardware family.

Shorter version: the HDA wall clock timer register interface didn't change at all in ACE, they just moved out of the crowded "shim" block to a timer-specific "TTS" block.